### PR TITLE
chore: update hyperview documentation

### PR DESCRIPTION
**Pull request by hgray-instawork**
Adding documentation around `<navigator>`, `<nav-route>` and `trigger="back"`

Asana: https://app.asana.com/0/1204008699308084/1205367406163854/f

### CODE REVIEW

File: docs/reference_behavior_attributes.md [hunk 1]
```diff
@@ -376,6 +387,8 @@ When using `new`, the fetched content must be either a full Hyperview XML docume
 
 If the element triggering the modal is a [`<form>`](/docs/reference_form) or contained in a `<form>` element, the form's data will be serialized as query parameters and included in the request. The form data has no effect when the href refers to a local `<screen>` element.
 
+<a id="action-back"></a>
```
Comment (adamstep) [inline docs/reference_behavior_attributes.md:390]: Is this needed?

File: docs/reference_index.md [hunk 2]
```diff
@@ -46,12 +46,19 @@ Input elements in HXML allow users to set local state on a Hyperview screen. Thi
...
 
 - [`<styles>`](/docs/reference_styles): Groups together all of the `<style>` rules for a screen.
 - [`<style>`](/docs/reference_style): A single style rule with unique id in the screen, can set multiple appearance properties.
 - [`<modifier>`](/docs/reference_modifier): Defines an appearance that should override the default for a style under certain local conditions, such as a user tap, selection, etc.
 
+#### Navigator Elements
+
+Navigator elements in HXML define a navigator hierarchy.
+
+- [`<navigator>`](/docs/reference_navigator): A definition of a stack or tab navigator.
```
Comment (adamstep) [inline docs/reference_index.md:59]: When I use the preview site, these links are 404s.

File: docs/reference_behavior_attributes.md (hunk 1 above)
Comment (hgray-instawork) [reply docs/reference_behavior_attributes.md:390]: It is. We now have two "back" sections. This anchor is needed to target the correct one in the links.

File: website/sidebars.json [hunk 3]
```diff
@@ -76,6 +76,10 @@
       "reference_hyperview_component",
       "reference_custom_elements",
       "reference_custom_behaviors"
+    ],
+    "Hyperview Navigation":[
```
Comment (adamstep) [inline website/sidebars.json:80]: For some reason, this section is not showing up in the sidebar on the preview site.

File: docs/reference_index.md (hunk 2 above)
Comment (hgray-instawork) [reply docs/reference_index.md:59]: Should be working now. You may need to run `yarn build` to see the updates.

File: website/sidebars.json (hunk 3 above)
Comment (hgray-instawork) [reply website/sidebars.json:80]: Should be working now. You may need to run `yarn build` to see the updates.

**Review by adamstep — APPROVED** [review]

**Review by flochtililoch — APPROVED** [review]

**Pull request merged**